### ISSUE TITLE
NPE when using withRegion(String region)

### ISSUE THREAD

**Issue by cscetbon** [aws/aws-sdk-java#2221]
Using [withRegion(Regions region)](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/client/builder/AwsClientBuilder.html#withRegion-com.amazonaws.regions.Regions-) works with no issue, however [withRegion(String region)](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/client/builder/AwsClientBuilder.html#withRegion-java.lang.String-) raises an NPE. Here is a sample kscript that reproduces the error:

```
#!/usr/bin/env kscript
//DEPS com.amazonaws:aws-java-sdk-dynamodb:1.11.719
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
import com.amazonaws.services.dynamodbv2.document.DynamoDB
import com.amazonaws.regions.Regions

val tableName = "my-table"
try {
    val dynamoDbClient = AmazonDynamoDBClientBuilder
        .standard()
        .withRegion(Regions.US_WEST_2.toString())
//        .withRegion(Regions.US_WEST_2.name)
//        .withRegion(Regions.US_WEST_2)
        .build()

    val dynamoDB = DynamoDB(dynamoDbClient)
    val tb = dynamoDB.getTable(tableName)
    println(tb.description)
    println(tb.scan().count())
    tb.scan().forEach {
        println(it.asMap())
    }
}
catch (e: Exception) {
    e.printStackTrace()
}
```
Only `.withRegion(Regions.US_WEST_2)` passes. Here is the stack trace:
```
null
java.lang.NullPointerException
	at java.lang.StringBuilder.<init>(StringBuilder.java:112)
	at com.amazonaws.auth.AWS4Signer.addHostHeader(AWS4Signer.java:547)
	at com.amazonaws.auth.AWS4Signer.sign(AWS4Signer.java:237)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1297)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1139)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:796)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:764)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:738)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:698)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:680)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:544)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:524)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.doInvoke(AmazonDynamoDBClient.java:5110)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:5077)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.executeScan(AmazonDynamoDBClient.java:3247)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.scan(AmazonDynamoDBClient.java:3213)
	at com.amazonaws.services.dynamodbv2.document.internal.ScanCollection.firstPage(ScanCollection.java:53)
	at com.amazonaws.services.dynamodbv2.document.internal.PageIterator.next(PageIterator.java:45)
	at com.amazonaws.services.dynamodbv2.document.internal.IteratorSupport.nextResource(IteratorSupport.java:87)
	at com.amazonaws.services.dynamodbv2.document.internal.IteratorSupport.hasNext(IteratorSupport.java:55)
	at kotlin.collections.CollectionsKt___CollectionsKt.count(_Collections.kt:1510)
	at Dynamodb_get_table.<init>(dynamodb-get-table.kts:21)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at Main_Dynamodb_get_table$Companion.main(Main_Dynamodb_get_table.kt:6)
	at Main_Dynamodb_get_table.main(Main_Dynamodb_get_table.kt)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jetbrains.kotlin.runner.AbstractRunner.run(runners.kt:64)
	at org.jetbrains.kotlin.runner.Main.run(Main.kt:110)
	at org.jetbrains.kotlin.runner.Main.main(Main.kt:120)
```

**Comment by cscetbon** [comment]
I found out that the name returned by `Regions.US_WEST_2.toString()` is `US_WEST_2` instead of `us-west-2`. So the sdk should say that the region does not exist instead of raising an NPE.

**Comment by debora-ito** [comment]
Hi @cscetbon I can confirm I'm seeing the NullPointerException, but this only happens if you're using SDK version `1.11.709` or newer. Until version 1.11.708 this would throw a IllegalArgumentException. 

I'll investigate what changed.

**Comment by debora-ito** [comment]
We identified the cause and a fix will bring back the IllegalArgumentException.

**Comment by cscetbon** [comment]
good to hear @debora-ito. Thanks

**Comment by debora-ito** [comment]
@cscetbon a fix was released in version `1.11.722` ([CHANGELOG](https://github.com/aws/aws-sdk-java/blob/master/CHANGELOG.md#111722-2020-02-13)). Please try it out and let us know if you have any other issue.
